azkaban-exec-server/src/main/java/azkaban/dag/DagBuilder.java

```diff
@@ -98,16 +141,16 @@ private void checkCircularDependencies() {
     class CircularDependencyChecker {
```

Comment:
The dependency check has already been done when loading the flow files: azkaban.project.DirectoryYamlFlowLoader#buildFlowEdges
Is there any particular reason why this needs to be implemented again in building the DAG?

Reply:
The DAG component is designed with minimal assumptions including the validations done to its inputs prior to passing these inputs to the DAG engine.

This allows this component to be used in more scenarios. e.g. the test included in this PR doesn't have a need to convert the v2 flow into Flow object which is where the azkaban.project.DirectoryYamlFlowLoader#buildFlowEdges is called. 

In the future, we can consider getting rid of one of the validations in the AZ loading scenario.  For the time being, the cost of double validation seems small.